lib/rules/jsx-quotes.js

```diff
@@ -19,11 +21,19 @@ var astUtils = require("../ast-utils");
 var QUOTE_SETTINGS = {
     "prefer-double": {
         quote: "\"",
-        description: "singlequote"
+        alternateQuote: "'",
+        description: "singlequote",
+        convert: function(str) {
+            return toDoubleQuotes(str);
```

Comment:
toDoubleQuotes doesn't work here, because jsx strings use xml escaping rules, where `"` needs to be converted to `&quot;` instead of `\"`